Add consistent WNP styling (fixes #10829)

maureenlholland opened the pull request:
## Description

Updates WNP styling to include a base partial that covers header, footer, and release notes styling shared across pages. Optional dark-mode styling available with import.

Changes to existing templates:
- remove main tag, header, and aside (these are now in base template)
- update content block to wnp_content

RESOLVED COMMENTS
- use optional dark mode import instead of including styling in base import (https://github.com/mozilla/bedrock/pull/10893#pullrequestreview-829038255)
- use white footer as default WNP footer styling (https://github.com/mozilla/bedrock/pull/10893#pullrequestreview-829001366)

SLACK DISCUSSION
- we can remove the experiments because they are finished
- we tend to keep up to 2 previous wnp releases because they can get a long tail of traffic, previous to fx93 can be removed

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/10829

related www-config PR to remove switches: https://github.com/mozmeao/www-config/pull/423

## Testing

existing pages using new base extension and/or updated styles (not an exhaustive list):

(light and dark mode)
http://localhost:8000/de/firefox/95.0/whatsnew/
http://localhost:8000/en-US/firefox/95.0/whatsnew/
(no dark mode)
http://localhost:8000/fr/firefox/93.0/whatsnew/
http://localhost:8000/en-US/firefox/93.0/whatsnew/
(nightly-specific)
http://localhost:8000/fr/firefox/94.0a1/whatsnew/

production (note: some of these will have the outdated firefox footer, new WNPs have white footer by default)
https://www.mozilla.org/de/firefox/95.0/whatsnew/
https://www.mozilla.org/en-US/firefox/95.0/whatsnew/
https://www.mozilla.org/fr/firefox/93.0/whatsnew/
https://www.mozilla.org/en-US/firefox/93.0/whatsnew/
https://www.mozilla.org/fr/firefox/94.0a1/whatsnew/

pages that should fall back to default whatsnew-account: https://www.mozilla.org/de/firefox/80.0/whatsnew/
Note: if signed-in, you should see mobile CTA, if signed-out, you should see Fx Accounts CTA
has send to device
http://localhost:8000/en-US/firefox/92.0/whatsnew/
has QR code
http://localhost:8000/it/firefox/91.0/whatsnew/

pages that should fall back to default whatsnew: https://www.mozilla.org/de/firefox/40.0/whatsnew/
has send to device
http://localhost:8000/en-US/firefox/40.0/whatsnew
has QR code
http://localhost:8000/it/firefox/40.0/whatsnew/

maureenlholland commented on media/css/firefox/whatsnew/includes/_header.scss:
```diff
@@ -46,11 +46,6 @@ $image-path: '/media/protocol/img';
     height: 40px;
     margin: 0;
-
-    @media (prefers-color-scheme: dark) {
```
needs to be added to old pages using this includes

maureenlholland commented on bedrock/firefox/templates/firefox/whatsnew/base.html:
```diff
@@ -13,3 +13,13 @@
+
+  {% block wnp_content %}{% endblock %}
+
+  {% include 'firefox/whatsnew/includes/release-notes.html' %}
```
needs removal from old wnp along with renaming of content block

maureenlholland commented on bedrock/firefox/templates/firefox/whatsnew/base-deprecated.html:
```diff
@@ -0,0 +1,15 @@
+{#
```
The idea here is to change the base extension in old WNPs to base-deprecated, and remove this file once all WNP before fx95 are deleted

maureenlholland commented on media/css/firefox/whatsnew/includes/_header-deprecated.scss:
```diff
@@ -0,0 +1,62 @@
+    height: 40px;
+    margin: 0;
+
+    @media (prefers-color-scheme: dark) {
```
Same idea as above here, change old WNP header style partials to use header-deprecated and delete once they are outdated. 

The other option is to move this header block into the old WNP individual stylesheets and not have a deprecated partial import.

maureenlholland commented on media/css/firefox/whatsnew/whatsnew-account.scss:
```diff
@@ -77,6 +77,11 @@ $image-path: '/media/protocol/img';
         color: $color-white;
```
this is combining two for dark mode blocks

maureenlholland commented:
TODO: 
- remove 93 exp switches: https://github.com/mozmeao/www-config/blob/master/waffle_configs/bedrock-prod.env#L41
- fix python wnp tests

maureenlholland commented on media/css/firefox/whatsnew/includes/_base.scss:
```diff
@@ -0,0 +1,8 @@
+@import 'header';
+@import 'footer';
+@import 'release-notes';
+@import 'dark-mode';
```
I'm on the fence about including this in the base and activating it with the `t-dark-mode` class in the template. 

Maybe it's better to leave out of the base and optionally import separately to the page's stylesheet? The would remove the need for a dark mode theme class in the template.

alexgibson replied:
I think this makes sense? 👍  

It could also be confusing to have both `t-dark` and `t-dark-mode` throughout various templates, when they do very different things?

maureenlholland replied:
I think moving this to an optional import is less bug-prone. Will update.

maureenlholland commented on media/css/firefox/whatsnew/includes/_footer.scss:
```diff
@@ -0,0 +1,64 @@
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// * -------------------------------------------------------------------------- */
+// Footer overrides
```
I kind of like using the white footer as a default WNP footer because it's really clean and a bit of a standout from the regular Firefox pages (with grey footer)

But I'm not sure how often it's actually used or if it should be an optional import (not included in base styles, allow inheritance of grey background firefox footer)

alexgibson replied:
Makes sesne to me, as the white footer styles have been the default on most WNPs for some time. We can always change it to be optional in the future if needed.

alexgibson commented on bedrock/firefox/views.py:
```diff
@@ -562,38 +548,17 @@ def get_template_names(self):
             template = "firefox/whatsnew/whatsnew-fx93-it.html"
         elif version.startswith("93.") and locale == "nl":
             template = "firefox/whatsnew/whatsnew-fx93-nl.html"
-        elif version.startswith("92.") and locale.startswith("en-"):
-            template = "firefox/whatsnew/whatsnew-fx92-en.html"
-        elif version.startswith("92.") and locale == "de":
-            template = "firefox/whatsnew/whatsnew-fx92-de.html"
-        elif version.startswith("92.") and locale == "fr":
-            template = "firefox/whatsnew/whatsnew-fx92-fr.html"
-        elif version.startswith("91.") and locale.startswith("en-"):
-            template = "firefox/whatsnew/whatsnew-fx91-en.html"
-        elif version.startswith("91.") and locale == "de":
-            template = "firefox/whatsnew/whatsnew-fx91-de.html"
```
Looks like there are functional tests for these templates that can be removed.

alexgibson commented on media/css/firefox/whatsnew/whatsnew-95-en.scss:
```diff
@@ -10,7 +10,8 @@ $image-path: '/media/protocol/img';
 @import '~@mozilla-protocol/core/protocol/css/includes/lib';
 @import '~@mozilla-protocol/core/protocol/css/templates/multi-column';
-@import 'includes/header';
+@import 'includes/base';
+@import 'includes/dark-mode';
```
Not a blocker, but I notice most of the pages get the default Firefox footer styles (hover states and bg color are different). Is there a reason why only WNP 95 gets the `base` and `dark-mode` imports?

I'm just wondering if we can simplify anything, so all templates get the same footer styles going forward?

maureenlholland replied:
I thought I'd leave existing styles and set up the base/dark-mode imports for future pages (with an example of use on fx95). But that was mainly because the PR was getting big. It should be fairly straightforward to make these updates across all remaining WNP templates. I think it's worth doing - thanks for the review!

alexgibson reviewed (changes requested):
Looks good - just a couple of test files that still need removing.

Also one non-blocking question about overall consistency.

maureenlholland commented:
@alexgibson ready for re-review! I deleted the mobile-de scss files because they were for 85/86 and don't seem to be in use anymore. Also deleted assets from the whatsnew/includes folder for same reason.

alexgibson reviewed (approved):
So much duplication removed, looks great nice work! r+

<img width="152" alt="image" src="https://user-images.githubusercontent.com/400117/145817669-3d1ee7d7-14e4-4c15-b1ed-2772170c7335.png">
